java-util/src/main/java/io/druid/java/util/common/RetryUtils.java

```diff
@@ -34,6 +33,16 @@
   public static final long MAX_SLEEP_MILLIS = 60000;
   public static final long BASE_SLEEP_MILLIS = 1000;
 
+  public interface Task<T>
+  {
+    T perform() throws Exception;
+  }
+
+  public interface OnFailure
```

Comment:
Maybe makes sense to be more specific and call it CleanupAfterFailure, and the method cleanupAfterFailure().

Reply:
Done.